packages/amplify-category-api/src/commands/api/update-secrets.ts

```diff
@@ -40,7 +37,6 @@ export const run = async (context: $TSContext) => {
   // read and validate the RDS connection parameters
   const databaseConfig: ImportedDataSourceConfig = await databaseConfigurationInputWalkthrough(engine, database);
 
-  await testDatabaseConnection(databaseConfig);
```

Comment:
If the database is in VPC, it would take several minutes to deploy a lambda and test the credentials which is not ideal if customers intent is to just update the credentials. So in order to keep the behavior consistent, the credentials are validated at generate schema step only.